easybuild/framework/easyblock.py

```diff
@@ -1428,6 +1428,26 @@ def make_module_extra(self, altroot=None, altversion=None):
                                      value, type(value))
             lines.append(self.module_generator.append_paths(key, value, allow_abs=self.cfg['allow_append_abs_path']))
 
+        # Add automatic PYTHONPATH or EBPYTHONPREFIXES if they aren't already present and python paths exist
+        if not os.path.isfile(f'{self.installdir}/bin/python'):  # only needed when not a base python installation
+            python_paths = [path[len(self.installdir)+1:]
+                            for path in glob.glob(f'{self.installdir}/lib/python*/site-packages')
+                            if re.match(self.installdir + r'/lib/python\d+\.\d+/site-packages', path)]
+
+            runtime_deps = [dep['name'] for dep in self.cfg.dependencies(runtime_only=True)]
+            use_ebpythonprefixes = 'Python' in runtime_deps and \
+                build_option('prefer_ebpythonprefixes') and self.cfg['prefer_ebpythonprefixes']
+
+            if python_paths:
+                # Add paths unless they were already added
+                if use_ebpythonprefixes:
+                    if '' not in self.module_generator.added_paths_per_key['EBPYTHONPREFIXES']:
```

Comment:
maybe worth a comment here why `''` makes sense (since it's relative to the installation prefix)

or introduce a local variable to help clarify what's going on here